pkg/devfile/adapters/helper.go

```diff
@@ -12,8 +12,8 @@ import (
 	"github.com/openshift/odo/pkg/odo/util/pushtarget"
 )
 
-// NewPlatformAdapter returns a Devfile adapter for the targeted platform
-func NewPlatformAdapter(componentName string, context string, devObj devfileParser.DevfileObj, platformContext interface{}) (PlatformAdapter, error) {
+// NewComponentAdapter returns a Devfile adapter for the targeted platform
```

Comment:
can we also rename the `ComponentAdapter` interface to `PlatformAdapter`?
https://github.com/openshift/odo/blob/7da08b69bab0cc6e172c4e1a39ac570566f26d1d/pkg/devfile/adapters/common/interface.go#L4

IMO  as this interface is intended for different platform(docker/k8s), `PlatformAdapter` sounds better, and component term is already too overloaded in odo.

Reply:
Agreed, I just kept the most used interface and its name but it's probably indeed better to rename it and maybe move it somewhere where it makes more sense.

Reply:
Actually, after taking another look at the code, `ComponentAdapter` is appropriate because its API deals with component and adapts the underlying implementation to the component functionality so I think it's OK to keep `ComponentAdapter`.

Reply:
but there is a component adapter present here https://github.com/openshift/odo/blob/5f71123d4332e47116b8c8112e76f91ec73f8833/pkg/devfile/adapters/kubernetes/component/adapter.go#L29
its not called componentAdapter but its inside component/adapter.go wouldn't that be confusing?

Reply:
There's also one here: https://github.com/openshift/odo/blob/5f71123d4332e47116b8c8112e76f91ec73f8833/pkg/devfile/adapters/docker/component/adapter.go#L21… What's your point? I didn't do an indepth refactoring here, just a quick win.